### PR TITLE
Improved dotests.sh output 

### PR DESCRIPTION
**Pull request by jubois**
- Changed output to diff --side-by-side style
- Runs all tests and set exit code to 0 only if all tests passed (else 1)

### CODE REVIEW

File: contrib/logtesting/dotests.sh [hunk 1]
```diff
@@ -19,17 +25,22 @@ for i in ./*/log; do
...
+        echo
+        echo
         echo "**ERROR: Unit testing failed. Output for the test $i failed."
-        echo "== OLD OUTPUT: =="
-        cat $idir/res
-	echo "== NEW OUTPUT: =="
-        cat tmpres
-        echo "** ERROR: Exiting."
+        echo "== DIFF OUTPUT (reference | new): =="
+        diff -Nay $idir/res tmpres
```
Comment (ddpbsd) [inline contrib/logtesting/dotests.sh]: My `diff` doesn't have a `-y` flag.

Comment (jubois) [reply contrib/logtesting/dotests.sh]: I checked with the virtual ossec appliance and the ``-y `` exists...
What is your distribution ?

Comment (ddpbsd) [reply contrib/logtesting/dotests.sh]: OpenBSD in this case. I haven't checked others.

**Comment by ddpbsd** [comment]
It doesn't work on OpenBSD anyway. I'll have to peek at that later. Thanks for the updates!

**Pull request merged**